### PR TITLE
FEI-4401.3: Cypress coverage reporting

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
This PR also simplifies out babel configuration.

Issue: FEI-4401

## Test plan:
- let GitHub actions do its thing
- check to see if coverage went up

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -105,13 +110,13 @@
...
-    "lint:timing": "TIMING=1 yarn lint",
+    "lint:timing": "cross-env TIMING=1 yarn lint",
     "publish:ci": "node utils/pre-publish-check-ci.js && git diff --stat --exit-code HEAD && yarn build && yarn extract-strings && changeset publish",
     "sloc": "sloc packages --exclude node_modules",
     "test": "yarn jest",
     "storybook": "start-storybook -p 6006",
     "build-storybook": "build-storybook",
     "cypress": "cypress open-ct",
-    "cypress:ci": "cypress run-ct"
+    "cypress:ci": "cross-env COVERAGE=1 cypress run-ct --env COVERAGE=1"
```
Comment (kevinbarabash) [inline package.json:120]: I wasn't able to replicate the original solution of add a config params to the babel-loader that could be passed to the babel config.  For some reason webpack didn't like the config options so here we are.  The `cross-env` environment variable is used by the babel config while the `--env` one is used by cypress.

File: .github/workflows/node-ci.yml [hunk 2]
```diff
@@ -164,32 +201,6 @@ jobs:
...
-    # Collect and record coverage
-    - name: Run Jest with coverage
-      run: yarn coverage
-    - name: Upload Coverage
-      # We don't bother collecting a record of coverage for dependabot changes
-      if: ${{ github.actor != 'dependabot[bot]' }}
-      uses: codecov/codecov-action@v2
-      with:
-        files: ./coverage/coverage-final.json
-        fail_ci_if_error: true
```
Comment (kevinbarabash) [inline .github/workflows/node-ci.yml:191]: I moved this section up to be next to the cypress coverage job.

File: package.json (hunk 1 above)
Comment (jeremywiebe) [reply package.json:120]: I wonder if we should name them differently so it's easier to differentiate which one is used where.  Like `BABEL_COVERAGE` and `CYPRESS_COVERAGE`?

**Comment by kevinbarabash** [comment]
The parent pull-request ([#249](https://github.com/Khan/perseus/pull/249)) has been merged into `main`, but this branch (`fei-4401.3`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Pull request merged**